File: cpp/src/ev1/ut2_controlflujometodos/u05repaso/ejercicios/resueltos/E10_RepasoIntegrado_Resuelto.cpp
/*
OBJETIVO: Menu 1=Tabla 5, 2=Par/Impar 4, 0=Salir con do-while.
SOLUCION: ver codigo.
PATRON: Menu do-while con opcion 0 salir y uso de cin.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
using namespace std;

void ImprimirMenu() {
    cout << endl;
    cout << "=== E10 Repaso Integrado Resuelto ===" << endl;
    cout << "1. Tabla del 5" << endl;
    cout << "2. Par o impar de 4" << endl;
    cout << "0. Salir" << endl;
}

int main() {
    int opcion;
    do {
        ImprimirMenu();
        cout << "Introduce una opcion -> ";
        cin >> opcion;
        switch (opcion) {
            case 1:
                for (int i = 1; i <= 10; i++)
                {
                    cout << "5 x " << i << " = " << (5 * i) << endl;
                }
                break;
            case 2:
                cout << (4 % 2 == 0 ? "Par" : "Impar") << endl;
                break;
            case 0:
                cout << "Saliendo..." << endl;
                break;
            default:
                cout << "Opcion no valida." << endl;
                break;
        }
    } while (opcion != 0);
    return 0;
}

File: csharp/src/ev2/ut5_pooexcepcionesio/u04repaso/ejercicios/resueltos/E10_RepasoIntegrado_Resuelto.cs
/*
OBJETIVO: Repaso integrado: clase Contacto, guardar en "contacto.txt" con try/catch IO. Menu do-while: mantener la solucion y ejecutarla desde menu interactivo.
SOLUCION: ver codigo.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

public class Program
{
    static void Main()
    {
        using System.IO;

        int opcion;

        do
        {
            ImprimirMenu();
            Console.Write("Introduce una opcion -> ");
            opcion = int.Parse(Console.ReadLine()!);

            switch (opcion)
            {
                case 1:
                    EjecutarEjercicio();
                    break;
                case 3:
                    EjecutarInteractivo();
                    break;
                case 2:
                    MostrarObjetivo();
                    break;
                case 0:
                    Console.WriteLine("Saliendo...");
                    break;
                default:
                    Console.WriteLine("Opcion no valida. Intenta de nuevo.");
                    break;
            }

            if (opcion != 0)
            {
                Console.WriteLine();
                Console.WriteLine("Pulsa ENTER para continuar...");
                Console.ReadLine();
                Console.Clear();
            }
        } while (opcion != 0);
    }

    static void ImprimirMenu()
    {
        Console.WriteLine("=== EJERCICIO ===");
        Console.WriteLine("1. Ejecutar solucion");
        Console.WriteLine("2. Ver objetivo");
        Console.WriteLine("3. Guardar contacto interactivo");
        Console.WriteLine("0. Salir");
        
    }

    static void MostrarObjetivo()
    {
        Console.WriteLine(@"Repaso integrado: clase Contacto, guardar en contacto.txt con try/catch IO.");
    }


    static void EjecutarEjercicio()
    {
        Contacto c = new Contacto("Laura", "laura@mail.com");
            try
            {
                File.WriteAllText("contacto.txt", c.ToLinea());
                Console.WriteLine("Contacto guardado en contacto.txt");
            }
            catch (IOException ex)
            {
                Console.WriteLine($"Error al escribir: {ex.Message}");
            }
    }

    static void EjecutarInteractivo()
    {
        Console.Write("Nombre: ");
        string? nombre = Console.ReadLine();
        Console.Write("Email: ");
        string? email = Console.ReadLine();
        Contacto c = new Contacto(nombre ?? "", email ?? "");
        try
        {
            File.WriteAllText("contacto.txt", c.ToLinea());
            Console.WriteLine("Contacto guardado en contacto.txt");
        }
        catch (IOException ex)
        {
            Console.WriteLine($"Error al escribir: {ex.Message}");
        }
    }
}


class Contacto
{
    public string Nombre { get; }
    public string Email { get; }

    public Contacto(string nombre, string email)
    {
        Nombre = nombre;
        Email = email;
    }

    public string ToLinea() => $"{Nombre};{Email}";
}
